packages/react-query/src/infiniteQueryOptions.ts

```diff
@@ -42,6 +42,7 @@ export type DefinedInitialDataInfiniteOptions<
   initialData:
     | NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>
     | (() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)
+    | undefined
```

Comment:
pretty sure this should go to `UndefinedInitialDataInfiniteOptions`, like we have for `queryOptions`. compare:

https://github.com/TanStack/query/blob/fdb8ce1c38bc30fec9e984218b2fdc7802aef695/packages/react-query/src/queryOptions.ts#L15

Reply:
Thankyou,
I just saw that you mentioned it in the issue. I'll reflect this.

But one thing what I wonder is, what is difference between two of them : 
- `(() => NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>)`
- `InitialDataFunction<NonUndefinedGuard<InfiniteData<TQueryFnData, TPageParam>>>`

Reply:
`InitialDataFunction` has `undefined` baked into the return value:

```
type InitialDataFunction<T> = () => T | undefined
```

Reply:
Thank you for your kindness!